packages/cxl-ui/src/components/cxl-marketing-nav.js

```diff
@@ -318,7 +362,11 @@ export class CXLMarketingNavElement extends LitElement {
 
         menuItem.appendChild(link);
 
+        // If item is at depth 1 and had no children, assume it's a section header, do not show description
+        if (item.sectionHeader && !item.children) {
```

Comment:
@freudFlintstone I need this to be lowercase `item.sectionheader` because tree JSON serializer we use in the PHP layer turns all keys into lowercase, I can't force it to `camelCase` and therefore section headers do not have background.

Reply:
Also it should not be limited to `a` component. I think section headers won't be of type `component => a`.

Reply:
About the component, it looks like Hener decided to use "component" to distinguish actual menu items from separators and "back" elements, which only appear in mobile screens. How about we ignore that property in the menu items? Maybe also change the hr to separator, so the spec would be:
```
"component" : 'back' | 'separator' | undefined  
```

Reply:
> About the component, I'm not 100% sure why it's implemented like that, but I tested a couple ideas and the it needs that to render the items correctly. I can look into it a little deeper tomorrow.

I do not agree. I just experimented and adding following code before `if (item.component === 'a')` will do what I want:
* add section header item,
* no link, just item.

```js
if (item.sectionheader) {
    const menuItem = document.createElement('vaadin-context-menu-item');
    menuItem.classList.add('section-header');
}
```

Is there anything wrong with adding context menu item without `<a>` inside @freudFlintstone @anoblet ?

Reply:
> But the <a> tag is created by cxl-marketing-nav and used in the menu list as a wrapper, so you are not obligated to pass only <a> components from your side. As long as it's valid HTML or plain text, it will be properly rendered.

At PHP level there's a condition which defines the component. Later in JS actual HTML element is created but as you can notice - there are conditions like:
* `if (item.component === 'a')`
* `if (item.component === 'back')`.

Maybe instead of working with `sectionheader` property we should define new component type?

Reply:
I'm fine with your solution for sectionheader, the difference in styling is too minimal to justify a different component just for that. 
For the `<a>` tag, no, it seems nothing that makes that mandatory, it was just a choice made years ago. It seems to be there just for creating links for items that have `href`, but it's use as the default for other items too, unless they are set to `hr` or `back`. 
Again, if checking for sectionheader before works, great, but we are probably still gonna need the `<a>` when href is set.

Reply:
There's still a small issue with that section label, because the figma spec uses a font-weight of 400, which we don't include in the current font. It is using a bolder 600 for now.
![image](https://github.com/conversionxl/aybolit/assets/1887825/6eb51c20-2509-472d-b22c-5b3d5a4237b8)